### PR TITLE
SLCORE-347 add branch parameter to ServerApi

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/IssueDownloader.java [hunk 1]
```diff
@@ -66,17 +67,19 @@ public IssueDownloader(IssueApi issueApi, SourceApi sourceApi, IssueStorePaths i
    * If the component doesn't exist or it exists but has no issues, an empty iterator is returned.
    *
    * @param key project key, or file key.
+   * @param branchName name of the branch. If null - issues will be downloaded for all branches.
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/IssueDownloader.java:70]: This is not how I understand the source code that handles `batch/issues`, I asked the SQ team about it and they confirmed that the default behavior (when `branch` is not set) is to return issues from the main branch only.

File: core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/perform/ServerIssueUpdater.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void update(String projectKey, Sonarlint.ProjectConfiguration projectConf
   }
 
   public void updateServerIssues(String projectKey, Sonarlint.ProjectConfiguration projectConfiguration, Path path, boolean fetchTaintVulnerabilities, ProgressWrapper progress) {
-    List<Sonarlint.ServerIssue> issues = issueDownloader.download(projectKey, projectConfiguration, fetchTaintVulnerabilities, progress);
+    List<Sonarlint.ServerIssue> issues = issueDownloader.download(projectKey, projectConfiguration, fetchTaintVulnerabilities, null, progress);
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/perform/ServerIssueUpdater.java:52]: Note: If I understand correctly, this is where we should inject the computed remote branch?

File: core/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java [hunk 3]
```diff
@@ -63,9 +63,12 @@ public IssueApi(ServerApiHelper serverApiHelper) {
    *
    * @param key project key, or file key.
    */
-  public DownloadIssuesResult downloadVulnerabilitiesForRules(String key, Set<String> ruleKeys, ProgressWrapper progress) {
+  public DownloadIssuesResult downloadVulnerabilitiesForRules(String key, Set<String> ruleKeys, @Nullable String branchName, ProgressWrapper progress) {
     StringBuilder searchUrl = new StringBuilder();
     searchUrl.append(getVulnerabilitiesUrl(key, ruleKeys));
+    if (branchName != null) {
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java:69]: Nitpick: the condition can be pushed to the builder method (lowers cognitive complexity).

File: core/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java [hunk 4]
```diff
@@ -106,12 +109,21 @@ private DownloadIssuesResult(List<Issue> issues, Map<String, Component> componen
...
+
+  private static String getVulnerabilitiesUrlBranchParameter(String branchName) {
+    return "&branch=" + urlEncode(branchName);
   }
 
-  public List<ScannerInput.ServerIssue> downloadAllFromBatchIssues(String key) {
+  public List<ScannerInput.ServerIssue> downloadAllFromBatchIssues(String key, @Nullable String branchName) {
+    var batchIssueUrl = new StringBuilder();
+    batchIssueUrl.append(getBatchIssuesUrl(key));
+    if (branchName != null) {
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/serverapi/issue/IssueApi.java:122]: Same comment as above.

**Review by jblievremont — APPROVED** [review]
Please see my few comments (especially the misleading Javadoc), else LGTM 👍🏻

File: core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/IssueDownloader.java (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/IssueDownloader.java:70]: Thanks. I'll change the javadoc.

File: core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/perform/ServerIssueUpdater.java (hunk 2 above)
Comment (kirill-knize-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/perform/ServerIssueUpdater.java:52]: Yes. And also we have a call from PartialUpdater. I'm not sure about it.

**Pull request merged**